moml/data/dataset_loader.py

```
@@ -0,0 +1,168 @@
+        datasets["val"] = load_dataset(val_dir, labels_file, file_pattern, config)
+
+    # Load test dataset (optional)
+    if os.path.isdir(test_dir):
+        datasets["test"] = load_dataset(test_dir, labels_file, file_pattern, config)
+
+    return datasets
+
+
+def load_hierarchical_dataset(
+    data_dir: str,
+    levels: List[str] = ["atom", "functional_group", "structural_motif"],
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)